src/gopher.cc

```diff
@@ -598,6 +598,9 @@ gopherToHTML(GopherStateData * gopherState, char *inbuf, int len)
 
                     } else if (gtype == GOPHER_INFO) {
                         snprintf(tmpbuf, TEMP_BUF_SIZE, "\t%s\n", html_quote(name));
+                    } else if (gtype == GOPHER_WWW) {	/* Gopher pointer to W3 */
+                        snprintf(tmpbuf, TEMP_BUF_SIZE, "<IMG border=\"0\" SRC=\"%s\"> <A HREF=\"%s\">%s</A>\n",
+                                     icon_url, selector, html_quote(name));
```

Comment:
It looks like similar `if` statements around the new code are using `escaped_selector` or even something more complex like `rfc1738_escape_unescaped(selector...)`. Please either document (via a C++ comment) why no `selector` escaping is necessary for `GOPHER_WWW` to prevent XSS/etc attacks OR adjust the code to use a properly escaped version of `selector`, whatever it is.

Reply:
Sure thing, the selector field for type w is a URL; it's probably a good idea to escape most special characters after the hostname part, I'll see what can be done.

Reply:
To be perfectly safe, the code should be using an AnyP::Uri object to parse() the URL received, and use absolute() accessor to get a normalized+escaped URL for the output only on successful parse.